modules/zone_association/variables.tf

```
@@ -0,0 +1,15 @@
+variable "vpc_id" {
+  type = object({
+    vpc_id = optional(string)
+  })
+  description = "VPC ID"
+  default = {
+    vpc_id = null
+  }
+}
```

We can simplify the input structure to match how it is used in the resource, removing the default since it is required for this resource.

```suggestion
variable "vpc_id" {
  description = "VPC ID"
  type = string
}
```